docs/choreolib/trajectory-api.md

```diff
@@ -0,0 +1,97 @@
+# Trajectory API
+
+!!! tip
+    Java users are recommended to use an [Auto Factory](./auto-factory.md) instead of the Trajectory API for loading and following trajectories, as it provides a more user-friendly experience for developing complex autonomous routines.
+
+To begin utilizing trajectories created by Choreo, you must first load a `Trajectory` ([Java](/api/choreolib/java/choreo/trajectory/Trajectory.html), [C++](/api/choreolib/cpp/classchoreo_1_1Trajectory.html)) from its corresponding file.
+
+=== "Java"
+
+    ```java
+    // Loads from deploy/choreo/myTrajectory.traj
+    // Optional.empty() is returned if the file does not exist or is invalid
+    var trajectory = Choreo.loadTrajectory("myTrajectory");
+    ```
+
+=== "C++"
+
+    ```cpp
+    // Loads from deploy/choreo/myTrajectory.traj
+    // std::nullopt is returned if the file does not exist or is invalid
+    auto trajectory = choreo::LoadTrajectory("myTrajectory")
+    ```
+
+=== "Python"
+
+    ```py
+    # Loads from deploy/choreo/myTrajectory.traj
+    # ValueError is thrown if the file does not exist or is invalid
+    try:
+      trajectory = choreo.load_swerve_trajectory("myTrajectory")
+    except ValueError:
+      # If the trajectory is not found, ChoreoLib already prints to DriverStation
+      pass
+    ```
+
+!!! warning
+    Trajectories should always be loaded at startup, not when the autonomous period begins. Loading trajectories is a blocking operation, and larger trajectories may take multiple seconds to load on a RoboRIO, cutting into the time a robot has to run its autonomous routine in a match.
```

Comment:
We should note that the Choreo AutoChooser's behaviour. The AutoChooser loads on select.